treewide: Remove Edge Impulse support and Machine Learning application (#29018)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/known_issues.rst

> nRF Machine Learning

Author: Should we somehow mark this section with an information that the application no longer exists?

Reviewer: Leave this file untouched in this PR. Instead, add comments/suggestions in my https://github.com/nrfconnect/sdk-nrf/pull/28979. And the answer to your question is no. We do not need to mark it. We might optionally have a mention saying that starting from NCS v3.4.0, this application is not part of the NCS, like, for example in the case of Amazon Sidewalk.

Author: Done, I'll add comments in the Known Issues PR.

Reviewer 2: Since there are reference link of the Machine learning application in the KI page, I think it would be good to remove the ref links in this PR to avoid the doc-build error.

Author: links replaced by plain text in known_issues.rst

Reviewer: Add the text about the removal of the application in NCS v3.4.0 and a link to the Add-on.

nRF Machine Learning
====================

The nRF Machine Learning application is removed in the |NCS| release v3.4.0 and replaced by the Gesture Recognition application in `Edge AI Add-on for nRF Connect SDK`_.

> The issues in this section are related to the nRF Machine Learning application from historical |NCS| releases.

Reviewer 2: ```suggestion
This section covers issues related to the nRF Machine Learning application found in previous |NCS| releases.
```
